be/src/storage/task/engine_batch_load_task.cpp

```diff
@@ -181,11 +208,14 @@ Status EngineBatchLoadTask::_push(const TPushReq& request, std::vector<TTabletIn
     Status res;
 
     if (request.tablet_type == TTabletType::TABLET_TYPE_LAKE) {
+        auto tablet_id = request.tablet_id;
+        auto tablet_version = (request.version == -1) ? choose_any_version(tablet_id) : request.version;
+        //                    ^^^^^^^^^^^^^^^^^^^^^^^ This should only happen when the FE version is less than 3.3.0
         auto tablet_manager = ExecEnv::GetInstance()->lake_tablet_manager();
-        auto tablet_or = tablet_manager->get_tablet(request.tablet_id, request.version);
+        auto tablet_or = tablet_manager->get_tablet(tablet_id, tablet_version);
```

Comment:
if choose_any_version returns invalid version, shall report error instead of let tablet_manager making an additional get request to remote and then error back.